src/backend/web/validation.js

```diff
@@ -19,53 +15,37 @@ const validate = (rules) => {
   };
 };
 
-// Expect only URI starting with http:// or https://
-const validateHttpSchemes = function (schema, uri) {
-  return typeof uri === 'string' && uri.search(/https?:\/\//i) === 0;
-};
-
-// Expect input fields not empty
-const validateNotEmpty = function (schema, data) {
-  return typeof data === 'string' && data.trim() !== '';
-};
-
-// Not Empty validation
-ajv.addKeyword('isNotEmpty', {
-  validate: validateNotEmpty,
-  errors: false,
-});
-
-// Http schemes validation
-ajv.addKeyword('httpSchemes', {
-  validate: validateHttpSchemes,
-  errors: false,
-});
-
-const feedSchema = {
-  type: 'object',
-  properties: {
-    user: { type: 'string', isNotEmpty: true },
-    url: {
-      type: 'string',
-      format: 'uri',
-      httpSchemes: true,
-    },
-    author: { type: 'string', isNotEmpty: true },
-  },
-};
-
-function validateNewFeed() {
-  return function (req, res, next) {
-    const feedData = req.body;
-    const valid = ajv.validate(feedSchema, feedData);
+// Rules for Validation of feeds for the /post route
+const feedSchemaRules = [
+  // user must be string and not empty
+  check('user')
+    .exists({ checkFalsy: true })
+    .withMessage('username is required')
+    .bail()
+    .isString()
+    .withMessage('username must be string')
+    .bail(),
+  // author must be string nad not empty
+  check('author')
+    .exists({ checkFalsy: true })
+    .withMessage('author is required')
+    .bail()
+    .isString()
+    .withMessage('author must be string')
+    .bail(),
+  // url must be string, follows url format and must be a valid url
+  check('url')
+    .isString()
+    .withMessage('url must be string')
+    .bail()
+    .isURL()
+    .withMessage('the format must be url')
+    .bail()
+    .matches(/https?:\/\//i)
```

Comment:
Let's use `isURL(str [, options])` and set `require_valid_protocol` to `true`

Reply:
what does the `str` refer to? If I understand the other part correctly, it would be like this, right?
```.isURL({ require_valid_protocol: true })```

Reply:
I tried to use both `protocols` and `required_valid_protocol` options but the test keeps failing. That's why, I used `match` separately. Also, I see that both of these are options are enabled by default. So, not sure why they don't work. Do you have any suggestions?

> options is an object which defaults to { protocols: ['http','https','ftp'], require_tld: true, require_protocol: false, require_host: true, require_valid_protocol: true, allow_underscores: false, host_whitelist: false, host_blacklist: false, allow_trailing_dot: false, allow_protocol_relative_urls: false, disallow_auth: false }.```

Reply:
Why is the test failing?  Is it allowing URLs with protocols other than http: and https:?  Can you limit it in the options?

Reply:
It's failing in these two lines and I think there's no protocol here
https://github.com/Seneca-CDOT/telescope/blob/master/test/feeds.test.js#L161-L168

Reply:
What does "failing" mean?  It accepts `www.invalidURL.com` and `chrome://www.google.com`?  Show me what your code looks like when it does.

Reply:
```javascript
check('url')
    .isString()
    .withMessage('url must be string')
    .bail()
    .isURL({ protocols: ['http', 'https'], require_valid_protocol: true })
    .withMessage('the format must be url')
    .bail(),
```
The Error:
<img width="805" alt="image" src="https://user-images.githubusercontent.com/64912621/99303107-5ed3c980-281e-11eb-867c-a61f3bab3f85.png">

Reply:
```
> const isUrl = require('validator/lib/isUrl');
undefined
> isUrl('www.yahoo.com')
true
> isUrl('www.yahoo.com', {require_protocol: true})
false
> isUrl('www.yahoo.com', {require_valid_protocol: true})
true
> isUrl('www.yahoo.com', {protocols: ['http', 'https'], require_valid_protocol: true})
true
> isUrl('www.yahoo.com', {protocols: ['http', 'https'], require_valid_protocol: true, require_protocol: true})
false
> isUrl('chrome://www.yahoo.com', {protocols: ['http', 'https'], require_valid_protocol: true, require_protocol: true})
false
```

So it looks like you need to use `{protocols: ['http', 'https'], require_valid_protocol: true, require_protocol: true}` for your options.

Reply:
OMG!! It works!! Thanks a lot :)